pkg/es/explain_allocation_test.go

```diff
@@ -0,0 +1,63 @@
+package es
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_elasticSearchClient_ExplainAllocation(t *testing.T) {
+	resp := `{
+	"error": {
+		"root_cause": [
+			{
+				"type": "illegal_argument_exception",
+				"reason": "No shard was specified in the request which means the response should explain a randomly-chosen unassigned shard, but there are no unassigned shards in this cluster. To explain the allocation of an assigned shard you must specify the target shard in the request."
+			}
+		],
+		"type": "illegal_argument_exception",
+		"reason": "No shard was specified in the request which means the response should explain a randomly-chosen unassigned shard, but there are no unassigned shards in this cluster. To explain the allocation of an assigned shard you must specify the target shard in the request."
```

Comment:
For now this will simply pass back the response from server,
this is a valid response as well the 400 case here. 

Later we can think of making this better UX.